addons/mail/models/mail_activity.py

```diff
@@ -330,13 +330,14 @@ def write(self, values):
         return res
 
     def unlink(self):
-        todo_activities = self.filtered(lambda act: act.date_deadline <= fields.Date.today())
+        only_active = self.filtered(lambda r: r.active)
+        todo_activities = only_active.filtered(lambda act: act.date_deadline <= fields.Date.today())
         if todo_activities:
             self.env['bus.bus']._sendmany([
                 [partner, 'mail.activity/updated', {'activity_deleted': True}]
                 for partner in todo_activities.user_id.partner_id
             ])
-        return super(MailActivity, self).unlink()
+        return super(MailActivity, only_active).unlink()
```

Comment:
![donenot_delete](https://github.com/user-attachments/assets/893f2b15-1790-449f-8481-c533ac93f43c)
Unfortunately, the 'done' filtered record is not deleted when we delete it from the actions.
 
 
 I did not get you why did you use the `self.filtered(lambda r: r.active)` in every method you have changed.

Reply:
Main reason behind adding that filter is to avoid any edge case situation where a user fabricates a way to display both done and pending activities in the same view. It will only consider the active fields and will perform the action for those records only.

Reply:
> Unfortunately, the 'done' filtered record is not deleted when we delete it from the actions.


You forget to implement this change :smiley:.

Reply:
The record(s) should be deleted regardless of their active state.